compiler/rustc_trait_selection/src/traits/dyn_compatibility.rs

```diff
@@ -598,8 +598,10 @@ fn receiver_is_dispatchable<'tcx>(
             ty::TraitRef::new_from_args(tcx, trait_def_id, args).upcast(tcx)
         };
 
-        let caller_bounds =
-            param_env.caller_bounds().iter().chain([unsize_predicate, trait_predicate]);
+        let caller_bounds = param_env
+            .caller_bounds()
+            .iter()
+            .chain(elaborate(tcx, [unsize_predicate, trait_predicate]));
```

Comment:
How come we don't call `normalize_param_env_or_error` on this `ParamEnv`, I would expect us to currently handle this check wrong when we have where clauses containing normalizable aliases, and after this change also handle supertraits of `U: Trait` wrong when they contain aliases.

Reply:
We could do that too, maybe i can make an error that requires the param env to be normalized